mlir/lib/IR/AffineExpr.cpp

```diff
@@ -237,6 +237,153 @@ unsigned AffineDimExpr::getPosition() const {
   return static_cast<ImplType *>(expr)->position;
 }
 
+/// Returns true if the expression is divisible by the given symbol with
+/// position `symbolPos`. The argument `opKind` signifies here which operation
+/// incorporates this division. It helps in implementing the commutative
+/// property of the floordiv and ceildiv operations. If the argument `exprKind`
+/// is floordiv and expr is also a binary expression of a floordiv operation,
+/// then the commutative property can be used otherwise, the floordiv operation
+/// is not divisible. The same argument is for the ceildiv operation.
+static bool isDivisibleBySymbol(AffineExpr expr, unsigned symbolPos,
+                                AffineExprKind opKind) {
```

Comment:
Shouldn't `opKind` be `divOpKind`? An accurate name is extremely important to understand - it makes the code self documenting! Isn't `opKind` always expected to be floordiv or ceildiv. You should really asserting that then at the beginning of the method. It's a pre-condition.

Reply:
1) It can be of three kind floordiv, ceildiv and modulo. If division check is because of modulo, then floordiv and ceildiv type of expressions are never divisible. 
Explaination
for the expression : (expr1 floordiv expr2) modulo expr3 and suppose expr3 is symbol s0. Now for its simplification I will check divisibility of (expr1 floordiv expr2) by s0 and if it is divisible then output is constant 0 otherwise can not be simplified. So function that would be called is indivisible by symbol((expr1 floordiv expr2), 0, modulo).
2) For expression like (expr1 floordiv expr2) can never be divided by symbol because of already existing division except for the case division is because of floordiv in which we can use commutative property. That is why we need to check which  operation is causing this division. 
I find it difficult to explain this much in comment form in code.